spark/src/main/java/org/apache/zeppelin/spark/SparkRInterpreter.java

```diff
@@ -0,0 +1,218 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.zeppelin.spark;
+
+import static org.apache.zeppelin.spark.ZeppelinRDisplay.render;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.apache.zeppelin.interpreter.*;
+import org.apache.zeppelin.scheduler.Scheduler;
+import org.apache.zeppelin.scheduler.SchedulerFactory;
+import org.jsoup.Jsoup;
+import org.jsoup.nodes.Document;
+import org.jsoup.nodes.Element;
+import org.jsoup.select.Elements;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Properties;
+
+/**
+ * R and SparkR interpreter with visualization support.
+ */
+public class SparkRInterpreter extends Interpreter {
+  private static final Logger logger = LoggerFactory.getLogger(SparkRInterpreter.class);
+
+  private static String renderOptions;
+
+  static {
+    Interpreter.register(
+      "r",
+      "spark",
+      SparkRInterpreter.class.getName(),
+      new InterpreterPropertyBuilder()
+              .add("spark.master",
+                      SparkInterpreter.getSystemDefault("MASTER", "spark.master", "local[*]"),
+                      "Spark master uri. ex) spark://masterhost:7077")
+              .add("spark.home",
+                      SparkInterpreter.getSystemDefault("SPARK_HOME", "spark.home", "/opt/spark"),
+                      "Spark distribution location")
```

Comment:
Why is this specifically need for the R interpreter.  SPARK_HOME is set globally in the zeppelin config prior to runtime.